src/components/Settings/SettingsAddressbookShare.vue

```diff
@@ -99,75 +88,38 @@ export default {
 		 * Share addressbook
 		 *
 		 * @param {Object} data destructuring object
-		 * @param {string} data.sharee the sharee
-		 * @param {string} data.id id
-		 * @param {Boolean} data.group group
+		 * @param {string} data.user the userId
+		 * @param {string} data.displayName the displayName
+		 * @param {string} data.uri the sharing principalScheme uri
+		 * @param {Boolean} data.isGroup is this a group ?
 		 */
-		shareAddressbook({ sharee, id, group }) {
+		shareAddressbook({ user, displayName, uri, isGroup }) {
 			let addressbook = this.addressbook
-			this.$store.dispatch('shareAddressbook', { addressbook, sharee, id, group })
-		},
-		/**
-		 * Format responses from axios.all and add them to the option array
-		 *
-		 * @param {Array} matches array of matches returned from the axios request
-		 * @param {String} query the search query
-		 * @param {Boolean} group Is this a group?
-		 */
-		formatMatchResults(matches, query, group) {
-			if (matches.length < 1) {
-				return
-			}
-			let regex = new RegExp(query, 'i')
-			let existingSharees = this.addressbook.shares.map(share => share.id + share.group)
-			matches = matches.filter(share => existingSharees.indexOf(share.id + group) === -1)
-			// this.usersOrGroups.concat(
-			this.usersOrGroups = this.usersOrGroups.concat(matches.map(match => {
-				let matchResult = match.displayname.split(regex)
-				if (matchResult.length < 1) {
-					return
-				}
-				return {
-					sharee: match.displayname,
-					id: match.id,
-					matchstart: matchResult[0],
-					matchpattern: match.displayname.match(regex)[0],
-					matchend: matchResult[1],
-					matchtag: group ? '(group)' : '(user)',
-					group
-				}
-			}))
+			this.$store.dispatch('shareAddressbook', { addressbook, user, displayName, uri, isGroup })
 		},
 
 		/**
 		 * Use Axios api call to find matches to the query from the existing Users & Groups
 		 *
 		 * @param {String} query
 		 */
-		asyncFind: debounce(function(query) {
+		findSharee: debounce(async function(query) {
 			this.isLoading = true
 			this.usersOrGroups = []
 			if (query.length > 0) {
-				api.all([
-					api.get(OC.linkToOCS('cloud', 2) + 'users/details?search=' + query),
-					api.get(OC.linkToOCS('cloud', 2) + 'groups/details?search=' + query)
-				]).then(response => {
-					let matchingUsers = Object.values(response[0].data.ocs.data.users)
-					let matchingGroups = response[1].data.ocs.data.groups
-					try {
-						this.formatMatchResults(matchingUsers, query, false)
-					} catch (error) {
-						console.debug(error)
+				const results = await client.principalPropertySearchByDisplayname(query)
```

Comment:
You have to filter the results to make sure you only get `calendarUserType` of either `INDIVIDUAL` or `GROUP`.
(Yes, all the results you see are either `INDIVIDUAL` or `GROUP`, but when using resource booking backends (which you probably are not), there will also be results of the `calendarUserType` `RESOURCE` and `ROOM`.)

Reply:
Oooh! Nice! Sorry I missed that :hugs: